gufe/tests/test_execute_dag.py

```diff
@@ -0,0 +1,106 @@
+# This code is part of gufe and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/gufe
+import os
+import pathlib
+import pytest
+import gufe
+from gufe.protocols import execute_DAG
+
+
+class WriterUnit(gufe.ProtocolUnit):
+    @staticmethod
+    def _execute(ctx, **inputs):
+        my_id = inputs['identity']
+
+        with open(os.path.join(ctx.shared, f'unit_{my_id}_shared.txt'), 'w') as out:
+            out.write(f'unit {my_id} existed!\n')
+        with open(os.path.join(ctx.scratch, f'unit_{my_id}_scratch.txt'), 'w') as out:
+            out.write(f'unit {my_id} was here\n')
+
+        return {
+            'log': 'finished',
+        }
```

Comment:
If the intent is to have a protocol unit that outputs a file to be saved, I think you'll need to include that as part of the return here.